src/zarr/store/local.py

```diff
@@ -187,20 +183,20 @@ async def set_partial_values(
         args = []
         for key, start, value in key_start_values:
             assert isinstance(key, str)
-            path = self.root / key
+            path = os.path.join(self.path, key)
             args.append((_put, path, value, start))
         await concurrent_map(args, to_thread, limit=None)  # TODO: fix limit
 
     async def delete(self, key: str) -> None:
         self._check_writable()
-        path = self.root / key
+        path = Path(self.path) / key
         if path.is_dir():  # TODO: support deleting directories? shutil.rmtree?
             shutil.rmtree(path)
         else:
             await to_thread(path.unlink, True)  # Q: we may want to raise if path is missing
 
     async def exists(self, key: str) -> bool:
-        path = self.root / key
+        path = Path(self.path) / key
```

Comment:
You do this a lot. I wonder if the store could have a private attr like `self._ppath = Path(self.path)` that does this once.

Reply:
I had a similar thought when I found myself doing this over and over again 😅

Reply:
done in https://github.com/zarr-developers/zarr-python/pull/2272/commits/c6c88435e8967c34f2bba2c529053897cae51798